### ISSUE TITLE
Errors in `InitGenesis`

### ISSUE THREAD

**Issue by whylee259** [CosmWasm/wasmd#335]
There exist errors in `InitGenesis` of the latest code.

1. Using params before they are initialized.
   - https://github.com/CosmWasm/wasmd/blob/master/x/wasm/internal/keeper/genesis.go#L17
      - https://github.com/CosmWasm/wasmd/blob/master/x/wasm/internal/keeper/keeper.go#L166
         - https://github.com/CosmWasm/wasmd/blob/master/x/wasm/internal/keeper/keeper.go#L117
   - This will cause a nil error.

2. The `LastCodeID` is always zero after importing codes
   - In `importCode`, It inserts codes but does not increase lastCodeId
   - So It always steps into the error case from the below code.
      - https://github.com/CosmWasm/wasmd/blob/master/x/wasm/internal/keeper/genesis.go#L47-L49

**Comment by alpe** [comment]
@whylee259 Thanks for reporting the 2 issues.

> 1. Using params before they are initialized.
You are completely right with this. Good 👁️ 

> 2. The LastCodeID is always zero after importing codes

🤔  I was not able to reproduce this issue. During the genesis code import the CodeID sequence is not incremented. Instead the sequences are seeded from the following genesis block later.
```json
  "wasm": {
      "sequences": [
        {
          "id_key": "BGxhc3RDb2RlSWQ=",
          "value": "2"
        },
        {
          "id_key": "BGxhc3RDb250cmFjdElk",
          "value": "2"
        }
      ]
...
We kept this separate to not run into issues for new instances when a derived contract account exists (accidentally). This scenario could happen when a contract instance was manually removed from the initial state dump for example but the account not. It is not obvious to find the relation between them so keeping the gap via sequences was a design decision.

**Comment by alpe** [comment]
@whylee259 This issue was closed now by #338 which addresses the first point only. If you still have issues with the second one, feel free to open this issue again or let us know via discord. 🌷
